sw/device/tests/crt_test.c

```diff
@@ -229,11 +232,12 @@ void _ottf_main(void) {
   CHECK(data_init_start_addr % sizeof(uint32_t) == 0,
         "_data_init_start not word-aligned: 0x%08x", data_init_start_addr);
 
-  CHECK(bad_bss_index == -1, "found non-zero .bss byte at *0x%08x == 0x%02x",
-        bss_start_addr + bad_bss_index, (uint32_t)bss[bad_bss_index]);
+  CHECK(bad_bss_index == -1,
+        "found non-zero .bss byte at *(0x%08x - 1) == 0x%02x", bss_start_addr,
+        (uint32_t)bss[bad_bss_index]);
```

Comment:
I definitely got confused when I made this change, so thank you for catching that.

Here are the errors that I'm trying to address.  Rather than doing all the necessary bounds checking to make the error message informative, I figure we can just omit the pointer arithmetic.

```
sw/device/tests/crt_test.c:236:26: error: implicit conversion changes signedness: 'int' to 'uintptr_t' (aka 'unsigned int') [-Werror,-Wsign-conversion]
        bss_start_addr + bad_bss_index, (uint32_t)bss[bad_bss_index]);
                       ~ ^~~~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:66:44: note: expanded from macro 'CHECK'
    if (status_err(CHECK_IMPL(condition, ##__VA_ARGS__))) { \
                                           ^~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:49:34: note: expanded from macro 'CHECK_IMPL'
        LOG_ERROR("CHECK-fail: " __VA_ARGS__);                             \
                                 ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:169:47: note: expanded from macro 'LOG_ERROR'
#define LOG_ERROR(...) LOG(kLogSeverityError, __VA_ARGS__)
                                              ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:124:30: note: expanded from macro 'LOG'
                           ##__VA_ARGS__); /* clang-format on */ \
                             ^~~~~~~~~~~
sw/device/tests/crt_test.c:236:26: error: implicit conversion changes signedness: 'int' to 'uintptr_t' (aka 'unsigned int') [-Werror,-Wsign-conversion]
        bss_start_addr + bad_bss_index, (uint32_t)bss[bad_bss_index]);
                       ~ ^~~~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:66:44: note: expanded from macro 'CHECK'
    if (status_err(CHECK_IMPL(condition, ##__VA_ARGS__))) { \
                                           ^~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:49:34: note: expanded from macro 'CHECK_IMPL'
        LOG_ERROR("CHECK-fail: " __VA_ARGS__);                             \
                                 ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:169:47: note: expanded from macro 'LOG_ERROR'
#define LOG_ERROR(...) LOG(kLogSeverityError, __VA_ARGS__)
                                              ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:128:44: note: expanded from macro 'LOG'
      base_log_internal_core(log_fields, ##__VA_ARGS__);         \
                                           ^~~~~~~~~~~
sw/device/tests/crt_test.c:239:27: error: implicit conversion changes signedness: 'int' to 'uintptr_t' (aka 'unsigned int') [-Werror,-Wsign-conversion]
        data_start_addr + bad_data_index, (uint32_t)data_init[bad_data_index]);
                        ~ ^~~~~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:66:44: note: expanded from macro 'CHECK'
    if (status_err(CHECK_IMPL(condition, ##__VA_ARGS__))) { \
                                           ^~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:49:34: note: expanded from macro 'CHECK_IMPL'
        LOG_ERROR("CHECK-fail: " __VA_ARGS__);                             \
                                 ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:169:47: note: expanded from macro 'LOG_ERROR'
#define LOG_ERROR(...) LOG(kLogSeverityError, __VA_ARGS__)
                                              ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:124:30: note: expanded from macro 'LOG'
                           ##__VA_ARGS__); /* clang-format on */ \
                             ^~~~~~~~~~~
sw/device/tests/crt_test.c:239:27: error: implicit conversion changes signedness: 'int' to 'uintptr_t' (aka 'unsigned int') [-Werror,-Wsign-conversion]
        data_start_addr + bad_data_index, (uint32_t)data_init[bad_data_index]);
                        ~ ^~~~~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:66:44: note: expanded from macro 'CHECK'
    if (status_err(CHECK_IMPL(condition, ##__VA_ARGS__))) { \
                                           ^~~~~~~~~~~
./sw/device/lib/testing/test_framework/check.h:49:34: note: expanded from macro 'CHECK_IMPL'
        LOG_ERROR("CHECK-fail: " __VA_ARGS__);                             \
                                 ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:169:47: note: expanded from macro 'LOG_ERROR'
#define LOG_ERROR(...) LOG(kLogSeverityError, __VA_ARGS__)
                                              ^~~~~~~~~~~
./sw/device/lib/runtime/log.h:128:44: note: expanded from macro 'LOG'
      base_log_internal_core(log_fields, ##__VA_ARGS__);         \
```